211 add chain id to package state

mohammadranjbarz opened the pull request:
resolve #211

mohammadranjbarz commented on .gitignore:
```diff
@@ -66,3 +66,6 @@ typings/
 # macOS
 .DS_Store
+
+
```
Thanks, Done.

mohammadranjbarz commented:
> Hey @mohammadranjbarz ! looks good, it is just also needed when executing `provision` steps during builds, here: https://github.com/usecannon/cannon/blob/211_add_chainId_to_package_state/packages/builder/src/steps/provision.ts/#L188

@mjlescano  Thanks for mentioning, I added it

dbeal-eth commented:
@mohammadranjbarz did you test running a build with this updated code and verify that the `chainId` is in fact included in the resulting cannon package on IPFS? just want to make sure we are in the habit of testing our changes in practice, at a minimum, before merging.

mohammadranjbarz commented:
> @mohammadranjbarz did you test running a build with this updated code and verify that the `chainId` is in fact included in the resulting cannon package on IPFS? just want to make sure we are in the habit of testing our changes in practice, at a minimum, before merging.

No I couldn't run it, because it uses typings imported from `@usecannon/builder` so although I changed typing of `DeploymentInfo` and added `chainId` but as it has not been published , my code tells you can pass `chainId` I thought we should merge it and after `@usecannon/builder` got published (assumed merging branch publish that package), we can test it

aminlatifi commented:
@mohammadranjbarz running `npm run build` on the `build` package will update compiled version which is imported by other packages. I have the env set up already on my machine, just please merge latest `main` branch  into yours

mohammadranjbarz commented:
> @mohammadranjbarz running `npm run build` on the `build` package will update compiled version which is imported by other packages. I have the env set up already on my machine, just please merge latest `main` branch into yours

Thanks for your feedback, done

aminlatifi commented:
I was curious whether it works, and found another bug! It doesn't work since the build process has a bug in publishing!
The URL is used for publishing to the IPFS is the one filled in the "ipfsUrl", not "publishIpfsUrl"

aminlatifi commented:
> I was curious whether it works, and found another bug! It doesn't work since the build process has a bug in publishing! The URL is used for publishing to the IPFS is the one filled in the "ipfsUrl", not "publishIpfsUrl"

Reported here
https://github.com/usecannon/cannon/issues/225

aminlatifi commented:
Tested, chainId was there :)
